hoomd/Trigger.h

```diff
@@ -87,5 +103,176 @@ class PYBIND11_EXPORT PeriodicTrigger : public Trigger
         uint64_t m_phase;
     };
 
+/** Until trigger
+ *
+ *  Trigger every time step until `timestep >= m_until`.
+*/
+class PYBIND11_EXPORT UntilTrigger : public Trigger
+    {
+    public:
+
+    UntilTrigger(uint64_t until) : Trigger(), m_until(until) {}
+
+    bool compute(uint64_t timestep)
+        {
+        if (timestep < m_until)
```

Comment:
Should we use <= here so that or(until(step),after(step)) is always true?

Reply:
Another way to do this would be to have `Before`, `On`, and `After` corresponding to `<`, `=`, and `>`. Then `or(before, on, after)` is always true. The language of "until" and "after" doesn't have a clear `=` to me.

Reply:
I like that idea. This was just a first pass for these kinds of triggering.